services/m3dbnode/server/serve.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package server
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/m3db/m3db/client"
+	"github.com/m3db/m3db/interfaces/m3db"
+	hjcluster "github.com/m3db/m3db/network/server/httpjson/cluster"
+	hjnode "github.com/m3db/m3db/network/server/httpjson/node"
+	ttcluster "github.com/m3db/m3db/network/server/tchannelthrift/cluster"
+	ttnode "github.com/m3db/m3db/network/server/tchannelthrift/node"
+	"github.com/m3db/m3db/sharding"
+	"github.com/m3db/m3db/storage"
+	"github.com/m3db/m3db/topology"
+
+	"github.com/spaolacci/murmur3"
+)
+
+func ShardingScheme() (m3db.ShardScheme, error) {
```

Comment:
I would actually perhaps just pass this into `Serve` so it can support different types of sharding schemes perhaps?  Either that or call it `DefaultShardingScheme`?